ugs-core/test/com/willwinder/universalgcodesender/model/GUIBackendTest.java

```diff
@@ -18,647 +18,357 @@ This file is part of Universal Gcode Sender (UGS).
-        fail("The test case is a prototype.");
-    }
+    public void pauseResumeWhenInIdleShouldThrowException() throws Exception {
+        when(controller.getControlState()).thenReturn(ControlState.COMM_IDLE);
+        instance.connect(FIRMWARE, PORT, BAUD_RATE);
 
-    /**
-     * Test of updateSystemState method, of class GUIBackend.
-     */
-    @Test
-    public void testUpdateSystemState() {
-        System.out.println("updateSystemState");
-        SystemStateBean systemStateBean = null;
-        GUIBackend instance = new GUIBackend();
-        instance.updateSystemState(systemStateBean);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
-    }
+        try {
+            instance.pauseResume();
+            fail("We should have had an exception here");
+        } catch (Exception e) {
+            // Expected!
+        }
 
-    /**
-     * Test of sendGcodeCommand method, of class GUIBackend.
-     */
-    @Test
-    public void testSendGcodeCommand() throws Exception {
-        System.out.println("sendGcodeCommand");
-        String commandText = "";
-        GUIBackend instance = new GUIBackend();
-        instance.sendGcodeCommand(commandText);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
+        verify(controller, times(0)).pauseStreaming();
+        verify(controller, times(0)).resumeStreaming();
     }
 
-    /**
-     * Test of adjustManualLocation method, of class GUIBackend.
-     */
     @Test
-    public void testAdjustManualLocation() throws Exception {
-        System.out.println("adjustManualLocation");
-        int dirX = 0;
-        int dirY = 0;
-        int dirZ = 0;
-        double stepSize = 0.0;
-        double feedRate = 0.0;
-        UnitUtils.Units units = null;
-        GUIBackend instance = new GUIBackend();
-        instance.adjustManualLocation(dirX, dirY, dirZ, stepSize, feedRate, units);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
-    }
+    public void pauseResumeWhenDisconnectedShouldThrowException() throws Exception {
+        when(controller.getControlState()).thenReturn(ControlState.COMM_DISCONNECTED);
+        instance.connect(FIRMWARE, PORT, BAUD_RATE);
 
-    /**
-     * Test of getSettings method, of class GUIBackend.
-     */
-    @Test
-    public void testGetSettings() {
-        System.out.println("getSettings");
-        GUIBackend instance = new GUIBackend();
-        Settings expResult = null;
-        Settings result = instance.getSettings();
-        assertEquals(expResult, result);
-        // TODO review the generated test code and remove the default call to fail.
-        fail("The test case is a prototype.");
-    }
+        try {
+            instance.pauseResume();
+            fail("We should have had an exception here");
```

Comment:
I like this, for some reason I always would set a `didThrow` boolean in the catch and verify later. I don't think it's a UGS dependency yet but recently I really like the assert4j method for this:
```java
assertThatThrownBy(() -> instance.pauseResume())
    .hasMessageContaining("whatever");
```